tensorflow/compiler/xla/service/gpu/fusion_bitcast_lift.cc

```diff
@@ -0,0 +1,259 @@
+          } else if (i->opcode() == HloOpcode::kParameter &&
+                     absl::c_all_of(i->users(), [](HloInstruction* u) {
+                         return u->opcode() == HloOpcode::kBitcast;
+                       })) {
+            VLOG(3) << "kParameter: " << i->ToString();
+            // Replace the parameter inside the fusion.
+            Shape new_shape = i->users()[0]->shape();
+            int64 parameter_number = i->parameter_number();
+            string name = i->name();
+            auto n = HloInstruction::CreateParameter(parameter_number,
+                                                     new_shape, name);
+            HloInstruction* new_parameter =
+              i->parent()->ReplaceParameter(parameter_number,
+                                            std::move(n));
+            // Remove the old inner bitcast.
+            auto old_users = new_parameter->users();
+            for (HloInstruction* param_user : old_users) {
+              DCHECK(param_user->opcode() == HloOpcode::kBitcast)
+                << "Expected a bitcast";
+              TF_RETURN_IF_ERROR(param_user->parent()->ReplaceInstructionWithDifferentShape(
+                  param_user, new_parameter));
+            }
+            // Replace the corresponding fusion operands with a new bitcast.
+            HloInstruction* old_outer_parameter =
+              cloned_fusion->mutable_operand(parameter_number);
+            HloInstruction* new_op =
+              old_outer_parameter->parent()->AddInstruction(
+                    HloInstruction::CreateBitcast(new_shape,
+                                                  old_outer_parameter));
+            TF_RETURN_IF_ERROR(cloned_fusion->ReplaceOperandWithDifferentShape(
+                parameter_number, new_op));
+            clone_changed = true;
+            changed = true;
+          } else if (i->opcode() == HloOpcode::kBroadcast) {
+            VLOG(3) << "kBroadcast: " << i->ToString();
+            // For now, do nothing. Later we can merge the broadcast
+            // and the bitcast, but this doesn't bring benefit in my
+            // current case.
+            if (set.count(i->mutable_operand(0)) == 0) {
+              stack.push_back(i->mutable_operand(0));
+              set.insert(i->mutable_operand(0));
+            }
+          } else if (i->opcode() == HloOpcode::kConstant &&
+                     !i->users().empty() &&
+                     absl::c_all_of(i->users(), [](HloInstruction* u) {
+                         return u->opcode() == HloOpcode::kBitcast;
+                       })) {
+            // Handling this case is optional for correctness, but
+            // handling it clean up the graph.
+            VLOG(3) << "kConstant: " << i->ToString();
+            Shape new_shape = i->users()[0]->shape();
+            TF_RETURN_IF_ERROR(i->parent()->ReplaceWithNewInstruction(
+                i, i->CloneWithNewOperands(new_shape, {})));
```

Comment:
@akuegel  Thanks for the update.
Does adding this line here fix the internal CI error?
`clone_changed = true;`
Here, we changed something.

Reply:
Yes, I have added this, and also changed = true, and it has already landed.